--- mozilla.orig/directory/c-sdk/ldap/include/Makefile.in	Thu Apr 11 16:01:14 2002
+++ mozilla/directory/c-sdk/ldap/include/Makefile.in	Fri Apr 19 07:15:31 2002
@@ -73,8 +73,6 @@
 	$(NSINSTALL) -D $(NSPRINCDIR)
 	$(INSTALL) $(INSTALLFLAGS) -m 644 $(HEADERS) $(INCLUDEDIR)
 	$(INSTALL) $(INSTALLFLAGS) -m 644 $(PRIVATEHEADERS) $(PRIVATEINCDIR)
-	rm -rf $(NSPRINCDIR)/*
-	cp -r $(DIST)/$(OBJDIR_NAME)/include/nspr/* $(NSPRINCDIR)
 
 $(INCLUDEDIR):	FORCE
 	$(NSINSTALL) -D $(INCLUDEDIR)
--- mozilla.orig/directory/c-sdk/ldap/libraries/libprldap/Makefile.in	Wed Apr 10 04:32:35 2002
+++ mozilla/directory/c-sdk/ldap/libraries/libprldap/Makefile.in	Fri Apr 19 07:03:01 2002
@@ -54,7 +54,7 @@
 
 LOCAL_INCLUDES  = -I$(PUBLIC)/nspr
 INCLUDES	+= -I$(DISTHDIR) -I$(HDIR) -I$(INSTALLDIR)/include \
-                   -I$(DIST)/include/nspr
+                   -I/usr/include/nspr
 DEFINES		+= $(DEFS)
 
 PLATFORMCFLAGS	= -DUSE_WAITPID -DNEEDPROTOS
