--- mozilla/xpfe/components/prefwindow/resources/content/preftree.xul.nodebug	Mon Jan  1 16:12:01 2001
+++ mozilla/xpfe/components/prefwindow/resources/content/preftree.xul	Mon Jan  1 16:13:39 2001
@@ -161,26 +161,6 @@
       </treechildren>
     </treeitem>
 
-    <treeitem container="true" open="true" id="debugItem">
-      <treerow>       
-        <treecell class="treecell-indent" url="chrome://communicator/content/pref/pref-debug.xul" value="&debug.label;"/> 
-      </treerow>
-      <treechildren id="debugChildren">
-
-        <treeitem>
-          <treerow>
-            <treecell class="treecell-indent" url="chrome://communicator/content/pref/pref-debug1.xul" value="&debug1.label;"/> 
-          </treerow>
-        </treeitem>
-        
-        <treeitem>
-          <treerow>
-            <treecell class="treecell-indent" url="chrome://communicator/content/pref/pref-debug2.xul" value="&debug2.label;"/> 
-          </treerow>
-        </treeitem>
-        
-      </treechildren>
-    </treeitem>
   </treechildren>
 
 </tree>
