--- mozilla/configure.in.orig	Mon May 27 21:29:12 2002
+++ mozilla/configure.in	Wed Aug 28 11:12:40 2002
@@ -4652,7 +4652,7 @@
         mkdir "directory/c-sdk/ldap"    
     fi
 
-    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --without-nss --with-mozilla"
+    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --with-nss --with-mozilla"
     if test -n "$HAVE_64BIT_OS"; then
         ac_configure_args="$ac_configure_args --enable-64bit"
     fi
--- mozilla/configure.orig	Mon May 27 21:29:00 2002
+++ mozilla/configure	Wed Aug 28 11:14:51 2002
@@ -14491,7 +14491,7 @@
         mkdir "directory/c-sdk/ldap"    
     fi
 
-    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --without-nss --with-mozilla"
+    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --with-nss --with-mozilla"
     if test -n "$HAVE_64BIT_OS"; then
         ac_configure_args="$ac_configure_args --enable-64bit"
     fi
--- mozilla/directory/c-sdk/ldap/libraries/libssldap/Makefile.in.orig	Fri May 17 08:54:00 2002
+++ mozilla/directory/c-sdk/ldap/libraries/libssldap/Makefile.in	Wed Aug 28 11:37:29 2002
@@ -53,9 +53,8 @@
 LOCAL_INCLUDES  = -I$(PUBLIC)/nspr
 INCLUDES	+= -I$(DISTHDIR) -I$(HDIR) -I$(INSTALLDIR)/include \
 		   -I$(DIST)/include \
-		   -I$(DIST)/public/security \
-		   -I$(DIST)/public/nss \
-                   -I$(DIST)/include/nspr
+		   -I/usr/include/nss \
+                   -I/usr/include/nspr
 DEFINES		+= $(DEFS)
 
 PLATFORMCFLAGS	= -DUSE_WAITPID -DNEEDPROTOS
