Index: widget/src/gtk2/nsFilePicker.cpp
===================================================================
RCS file: /cvsroot/mozilla/widget/src/gtk2/nsFilePicker.cpp,v
retrieving revision 1.4
diff -d -u -p -r1.4 nsFilePicker.cpp
--- widget/src/gtk2/nsFilePicker.cpp	22 Sep 2004 04:30:49 -0000	1.4
+++ widget/src/gtk2/nsFilePicker.cpp	27 Sep 2004 08:12:30 -0000
@@ -43,7 +43,7 @@
 #include "nsIURI.h"
 #include "nsIWidget.h"
 #include "nsILocalFile.h"
-#include "nsISupportsArray.h"
+#include "nsArrayEnumerator.h"
 #include "nsVoidArray.h"
 #include "nsMemory.h"
 #include "nsEnumeratorUtils.h"
@@ -80,6 +80,7 @@ typedef enum
 
 
 typedef gchar* (*_gtk_file_chooser_get_filename_fn)(GtkFileChooser *chooser);
+typedef GSList* (*_gtk_file_chooser_get_filenames_fn)(GtkFileChooser *chooser);
 typedef GtkWidget* (*_gtk_file_chooser_dialog_new_fn)(const gchar *title,
                                                       GtkWindow *parent,
                                                       GtkFileChooserAction action,
@@ -95,6 +96,7 @@ typedef void (*_gtk_file_filter_set_name
 
 
 DECL_FUNC_PTR(gtk_file_chooser_get_filename);
+DECL_FUNC_PTR(gtk_file_chooser_get_filenames);
 DECL_FUNC_PTR(gtk_file_chooser_dialog_new);
 DECL_FUNC_PTR(gtk_file_chooser_set_select_multiple);
 DECL_FUNC_PTR(gtk_file_chooser_set_current_name);
@@ -148,6 +150,7 @@ nsFilePicker::LoadSymbolsGTK24()
     GET_LIBGTK_FUNC(gtk_file_chooser_get_filename);
   }
 
+  GET_LIBGTK_FUNC(gtk_file_chooser_get_filenames);
   GET_LIBGTK_FUNC(gtk_file_chooser_dialog_new);
   GET_LIBGTK_FUNC(gtk_file_chooser_set_select_multiple);
   GET_LIBGTK_FUNC(gtk_file_chooser_set_current_name);
@@ -212,12 +215,36 @@ nsFilePicker::~nsFilePicker()
 }
 
 void
-nsFilePicker::ReadValuesFromFileChooser(GtkWidget *file_chooser)
+ReadMultipleFiles(gpointer filename, gpointer array)
 {
-  // Grab the filename
-  gchar *filename = _gtk_file_chooser_get_filename (GTK_FILE_CHOOSER(file_chooser));
-  mFile.Assign(filename);
+  nsCOMPtr<nsILocalFile> localfile;
+  nsresult rv = NS_NewNativeLocalFile(nsDependentCString(NS_STATIC_CAST(char*, filename)),
+                                      PR_FALSE,
+                                      getter_AddRefs(localfile));
+  if (NS_SUCCEEDED(rv)) {
+    nsCOMArray<nsILocalFile>& files = *NS_STATIC_CAST(nsCOMArray<nsILocalFile>*, array);
+    files.AppendObject(localfile);
+  }
+
   g_free(filename);
+}
+
+void
+nsFilePicker::ReadValuesFromFileChooser(GtkWidget *file_chooser)
+{
+  mFiles.Clear();
+
+  if (mMode == nsIFilePicker::modeOpenMultiple) {
+    mFile.Truncate();
+
+    GSList *list = _gtk_file_chooser_get_filenames (GTK_FILE_CHOOSER(file_chooser));
+    g_slist_foreach(list, ReadMultipleFiles, NS_STATIC_CAST(gpointer, &mFiles));
+    g_slist_free(list);
+  } else {
+    gchar *filename = _gtk_file_chooser_get_filename (GTK_FILE_CHOOSER(file_chooser));
+    mFile.Assign(filename);
+    g_free(filename);
+  }
 
   // Remember last used directory.
   nsCOMPtr<nsILocalFile> file;
Index: widget/src/gtk2/nsFilePicker.h
===================================================================
RCS file: /cvsroot/mozilla/widget/src/gtk2/nsFilePicker.h,v
retrieving revision 1.2
diff -d -u -p -r1.2 nsFilePicker.h
--- widget/src/gtk2/nsFilePicker.h	8 Aug 2004 11:09:51 -0000	1.2
+++ widget/src/gtk2/nsFilePicker.h	27 Sep 2004 08:12:30 -0000
@@ -43,10 +43,10 @@
 #include "nsBaseFilePicker.h"
 #include "nsString.h"
 #include "nsVoidArray.h"
+#include "nsCOMArray.h"
 
 class nsIWidget;
 class nsILocalFile;
-class nsISupportsArray;
 class PRLibrary;
 
 class nsFilePicker : public nsBaseFilePicker
@@ -70,7 +70,7 @@ protected:
 
   nsCOMPtr<nsIWidget>    mParentWidget;
   nsCOMPtr<nsILocalFile> mDisplayDirectory;
-  nsCOMPtr<nsISupportsArray> mFiles;
+  nsCOMArray<nsILocalFile> mFiles;
 
   PRInt16   mMode;
   PRInt16   mSelectedType;
