--- mozilla/nsprpub/configure.in.libc	Sun Apr 22 14:18:48 2001
+++ mozilla/nsprpub/configure.in	Sun Apr 22 14:23:08 2001
@@ -886,6 +886,7 @@
     MKSHLIB='$(LD) $(DSO_LDOPTS) -soname $(notdir $@) -o $@'
     DSO_CFLAGS=-fPIC
     DSO_LDOPTS=-shared
+    OS_LIBS="$OS_LIBS -lc"
     case "${target_cpu}" in
     alpha)
         AC_DEFINE(_ALPHA_)
--- mozilla/configure.in.libc	Sun Apr 22 14:18:42 2001
+++ mozilla/configure.in	Sun Apr 22 14:22:45 2001
@@ -725,6 +725,7 @@
     AC_DEFINE(_POSIX_SOURCE)
     AC_DEFINE(_SVID_SOURCE)
     TARGET_NSPR_MDCPUCFG='\"md/_linux.cfg\"'
+    LIBS="$LIBS -lc"
 
     case "${host_cpu}" in
     alpha*)
