--- mozilla/security/nss/lib/util/unix_rand.c.screw	Fri Feb 16 13:20:41 2001
+++ mozilla/security/nss/lib/util/unix_rand.c	Fri Feb 16 13:20:52 2001
@@ -263,7 +263,6 @@
 #endif /* IBM R2 */
 
 #if defined(LINUX)
-#include <linux/kernel.h>
 
 static size_t
 GetHighResClock(void *buf, size_t maxbytes)
--- mozilla/security/nss/lib/manifest.mn.screw	Fri Feb 16 13:22:18 2001
+++ mozilla/security/nss/lib/manifest.mn	Fri Feb 16 13:22:29 2001
@@ -36,7 +36,7 @@
 DIRS = pkcs7 ssl nss crmf jar \
     certhigh pk11wrap cryptohi \
     softoken certdb crypto \
-    util freebl pkcs12 fortcrypt \
+    freebl util pkcs12 fortcrypt \
     smime base ckfw
 #
 # these dirs are not built at the moment
diff -Nrc -x *~ mozilla.orig/security/coreconf/headers.mk mozilla/security/coreconf/headers.mk
*** mozilla.orig/security/coreconf/headers.mk	Mon Nov 20 19:01:53 2000
--- mozilla/security/coreconf/headers.mk	Thu Apr  5 21:16:24 2001
***************
*** 52,54 ****
--- 52,59 ----
  #
  
  INCLUDES += -I$(SOURCE_XPPRIVATE_DIR)
+ 
+ ifdef NSPR_INCLUDEDIR
+ INCLUDES += -I$(NSPR_INCLUDEDIR)
+ endif
+ 
--- mozilla.orig/security/psm/server/Makefile	Wed Dec 20 01:24:58 2000
+++ mozilla/security/psm/server/Makefile	Fri Feb 16 07:30:47 2001
@@ -144,9 +144,9 @@
 
 # Removing Hardcoded shared lib suffixes so that HP-UX Builds
 NSPR_LIBS = \
-	$(DIST)/lib/libplc4.$(DLL_SUFFIX) \
-	$(DIST)/lib/libplds4.$(DLL_SUFFIX) \
-	$(DIST)/lib/libnspr4.$(DLL_SUFFIX)
+	-lplc4 \
+	-lplds4 \
+	-lnspr4
 
 XPCOM_LIBS = \
 	$(DIST)/lib/libxpcom.$(DLL_SUFFIX) \
