--- mozilla/extensions/psm-glue/src/nsPSMUICallbacks.cpp.open	Wed Oct 18 11:59:02 2000
+++ mozilla/extensions/psm-glue/src/nsPSMUICallbacks.cpp	Wed Oct 18 11:59:10 2000
@@ -106,7 +106,11 @@
   char buffer[256];
   PR_snprintf(buffer,
               sizeof(buffer),
-              modal ? "menubar=no,height=%d,width=%d,dependent,modal"
+#ifdef WIN32
+	       modal ? "menubar=no,height=%d,width=%d,dependent,modal"
+#else
+              (modal && win) ? "menubar=no,height=%d,width=%d,dependent"
+#endif
               : "menubar=no,height=%d,width=%d",
               height,
               width );
@@ -115,12 +119,16 @@
   if (argv) {
     // open the window
     nsIDOMWindowInternal *newWindow;
+#ifdef WIN32 
     if (modal && win) {
       parentWindow->OpenDialog(jsContext, argv, 3, &newWindow);
     } else {
       parentWindow->Open(jsContext, argv, 3, &newWindow);
     }
     newWindow->ResizeTo(width, height);
+#else
+    parentWindow->Open(jsContext, argv, 3, &newWindow);
+#endif
     JS_PopArguments(jsContext, stackPtr);
   }
  loser:
