--- mozilla/gfx/src/gtk/nsRenderingContextGTK.cpp.orig	Wed Nov 28 16:34:04 2001
+++ mozilla/gfx/src/gtk/nsRenderingContextGTK.cpp	Wed Nov 28 15:39:05 2001
@@ -1980,6 +1980,10 @@
                                          const gchar *text,
                                          gint         text_length)
 {
+
+#define MOZ_DRAW_TEXT_WITH_GDK 
+#ifndef MOZ_DRAW_TEXT_WITH_GDK /* gdkxft patch */
+
   GdkWindowPrivate *drawable_private;
   GdkFontPrivate *font_private;
   GdkGCPrivate *gc_private;
@@ -2026,6 +2030,12 @@
   }
   else
     g_error("undefined font type\n");
+
+#else /* MOZ_DRAW_TEXT_WITH_GDK */
+
+  gdk_draw_text(drawable, font, gc, x, y, text, MIN(text_length, 32768));
+
+#endif /* MOZ_DRAW_TEXT_WITH_GDK */
 }
 
 
