--- mozilla/security/manager/Makefile.in.orig	Fri Mar  8 23:50:11 2002
+++ mozilla/security/manager/Makefile.in	Sat Apr 13 16:07:59 2002
@@ -95,14 +95,6 @@
 else
 	cd $(DIST)/lib; cp -f $(LIB_PREFIX)mozdbm_s.$(LIB_SUFFIX) $(LIB_PREFIX)dbm.$(LIB_SUFFIX); $(RANLIB) $(LIB_PREFIX)dbm.$(LIB_SUFFIX)
 endif
-	$(MAKE) -C $(topsrcdir)/security/nss/lib $(DEFAULT_GMAKE_FLAGS)
-	$(INSTALL) -m 755 $(DIST)/lib/$(LOADABLE_ROOT_MODULE) $(DIST)/bin
-ifdef NSS_3_4
-	$(INSTALL) -m 755 $(DIST)/lib/$(SOFTOKEN3_LIB) $(DIST)/bin
-	$(INSTALL) -m 755 $(DIST)/lib/$(NSS3_LIB) $(DIST)/bin
-	$(INSTALL) -m 755 $(DIST)/lib/$(SSL3_LIB) $(DIST)/bin
-	$(INSTALL) -m 755 $(DIST)/lib/$(SMIME3_LIB) $(DIST)/bin
-endif
 ifneq (,$(filter SunOS HP-UX,$(OS_ARCH)))
 ifneq ($(OS_TEST),i86pc)
 ifndef HAVE_64BIT_OS
--- mozilla/security/manager/ssl/src/Makefile.in.orig	Tue Feb 19 04:00:08 2002
+++ mozilla/security/manager/ssl/src/Makefile.in	Sat Apr 13 17:15:18 2002
@@ -108,8 +108,7 @@
 
 # Use local includes because they are inserted before INCLUDES
 # so that Mozilla's nss.h is used, not glibc's
-LOCAL_INCLUDES += -I$(DIST)/public/security \
-                  -I$(DIST)/private/security \
+LOCAL_INCLUDES += -I/usr/include/nss \
                   $(NULL)
 
 EXTRA_DSO_LDOPTS += \
--- mozilla/config/config.mk.orig	Wed Mar  6 10:56:33 2002
+++ mozilla/config/config.mk	Sat Apr 13 18:29:08 2002
@@ -174,7 +174,7 @@
 else
 NSS_LIBS	= \
 	$(LIBS_DIR) \
-	$(DIST)/lib/$(LIB_PREFIX)crmf.$(LIB_SUFFIX) \
+	/usr/lib/libcrmf.a \
 	-lsmime3 \
 	-lssl3 \
 	-lnss3 \
@@ -183,11 +183,6 @@
 endif
 
 NSS_DEP_LIBS	= \
-	$(DIST)/lib/$(LIB_PREFIX)crmf.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)smime3$(DLL_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)ssl3$(DLL_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)nss3$(DLL_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)softokn3$(DLL_SUFFIX) \
 	$(NULL)
 
 MOZ_UNICHARUTIL_LIBS = $(DIST)/lib/$(LIB_PREFIX)unicharutil_s.$(LIB_SUFFIX)
