diff -u -r1.804 configure.in
--- mozilla/configure.in	2001/03/25 20:18:18	1.804
+++ mozilla/configure.in	2001/03/26 09:37:37
@@ -3347,11 +3347,8 @@
 MOZ_ARG_WITH_STRING(default-mozilla-five-home,
 [  --with-default-mozilla-five-home    Set the default value for MOZILLA_FIVE_HOME],
   [ val=`echo $withval`
-    DSO_LDOPTS="$DSO_LDOPTS -Wl,-rpath=$val"
-    LIBS="$LIBS -Wl,-rpath=$val"
     AC_DEFINE_UNQUOTED(MOZ_DEFAULT_MOZILLA_FIVE_HOME,"$val")
-    MOZ_NSPRENV_OVERRIDE_DSO_LDOPTS="-shared -rpath=$val"
-    AC_SUBST(MOZ_NSPRENV_OVERRIDE_DSO_LDOPTS) ] )
+    ])
 dnl ========================================================
 
 dnl ========================================================
