--- configure.in.orig	Sat Sep 30 23:16:13 2000
+++ configure.in	Tue Oct 17 16:00:58 2000
@@ -3114,6 +3114,25 @@
 dnl =   pass args to --without-debug-modules)
 dnl = 
 dnl ========================================================
+MOZ_ARG_WITH_STRING(default-mozilla-five-home,
+[  --with-default-mozilla-five-home    Set the default value for MOZILLA_FIVE_HOME],
+  [ val=`echo $withval`
+    DSO_LDOPTS="$DSO_LDOPTS -Wl,-rpath=$val"
+    LIBS="$LIBS -Wl,-rpath=$val"
+    AC_DEFINE_UNQUOTED(MOZ_DEFAULT_MOZILLA_FIVE_HOME,"$val")
+    MOZ_NSPRENV_OVERRIDE_DSO_LDOPTS="-shared -rpath=$val"
+    AC_SUBST(MOZ_NSPRENV_OVERRIDE_DSO_LDOPTS) ] )
+dnl ========================================================
+
+dnl ========================================================
+dnl =
+dnl = --with-debug-modules=module1,^module2,etc
+dnl = 
+dnl = Enable/disable debug for specific modules only
+dnl =   module names beginning with ^ will be disabled (this is because you can't
+dnl =   pass args to --without-debug-modules)
+dnl = 
+dnl ========================================================
 
 MOZ_ARG_WITH_STRING(debug-modules,
 [  --with-debug-modules    Enable/disable debug for specific modules only],
--- build/unix/nspr_my_overrides.mk.in.orig	Thu Apr 15 19:38:58 1999
+++ build/unix/nspr_my_overrides.mk.in	Tue Oct 17 16:00:58 2000
@@ -1,7 +1,12 @@
 DIST=@MOZ_NSPRENV_DIST@
 MOZ_NSPRENV_OVERRIDE_MAKE=@MOZ_NSPRENV_OVERRIDE_MAKE@
+MOZ_NSPRENV_OVERRIDE_DSO_LDOPTS=@MOZ_NSPRENV_OVERRIDE_DSO_LDOPTS@
 
 ifdef MOZ_NSPRENV_OVERRIDE_MAKE
 MAKE=$(MOZ_NSPRENV_OVERRIDE_MAKE)
+endif
+
+ifdef MOZ_NSPRENV_OVERRIDE_DSO_LDOPTS
+DSO_LDOPTS=$(MOZ_NSPRENV_OVERRIDE_DSO_LDOPTS)
 endif
 
--- xpcom/io/nsDirectoryService.cpp.orig	Fri Sep 29 13:40:10 2000
+++ xpcom/io/nsDirectoryService.cpp	Tue Oct 17 16:00:58 2000
@@ -196,7 +196,26 @@
     //	- if MOZILLA_FIVE_HOME is defined, that is it
     //	- else give the current directory
     char buf[MAXPATHLEN];
+
+    // The MOZ_DEFAULT_MOZILLA_FIVE_HOME variable can be set at configure time with
+    // a --with-default-mozilla-five-home=foo autoconf flag.
+    // 
+    // The idea here is to allow for builds that have a default MOZILLA_FIVE_HOME
+    // regardless of the environment.  This makes it easier to write apps that
+    // embed mozilla without having to worry about setting up the environment 
+    //
+    // We do this py putenv()ing the default value into the environment.  Note that
+    // we only do this if it is not already set.
+#ifdef MOZ_DEFAULT_MOZILLA_FIVE_HOME
+    if (PR_GetEnv("MOZILLA_FIVE_HOME") == nsnull)
+    {
+	    putenv ("MOZILLA_FIVE_HOME=" MOZ_DEFAULT_MOZILLA_FIVE_HOME);
+    }
+#endif
+
     char *moz5 = PR_GetEnv("MOZILLA_FIVE_HOME");
+
+
     if (moz5)
     {
         localFile->InitWithPath(moz5);
