--- mozilla/security/coreconf/Linux.mk.sparc-config	Thu Dec 28 17:18:52 2000
+++ mozilla/security/coreconf/Linux.mk	Thu Dec 28 17:20:01 2000
@@ -58,8 +58,18 @@
         OS_REL_CFLAGS   = -D_ALPHA_ -DLINUX1_2 -D_XOPEN_SOURCE
 	CPU_ARCH	= alpha
 else
+ifeq ($(OS_TEST),sparc)
+	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+	CPU_ARCH        = sparc
+else
+ifeq ($(OS_TEST),sparc64)
+	OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+	CPU_ARCH        = sparc
+else
 	OS_REL_CFLAGS	= -DLINUX1_2 -Di386 -D_XOPEN_SOURCE
 	CPU_ARCH	= x86
+endif
+endif
 endif
 endif
 
