diff -urN mozilla.orig/config/config.mk mozilla/config/config.mk
--- mozilla.orig/config/config.mk	Sun Dec  9 08:08:58 2001
+++ mozilla/config/config.mk	Sat Jan 26 23:17:12 2002
@@ -164,9 +164,8 @@
 NSS_LIBS	= \
 	$(DIST)/lib/libsmime.$(LIB_SUFFIX) \
 	$(DIST)/lib/libcrmf.$(LIB_SUFFIX) \
-	$(DIST)/lib/libssl.$(LIB_SUFFIX) \
+	/usr/lib/libssl3.a \
 	$(DIST)/lib/libnss.$(LIB_SUFFIX) \
-	$(DIST)/lib/libssl.$(LIB_SUFFIX) \
 	$(DIST)/lib/libpkcs12.$(LIB_SUFFIX) \
 	$(DIST)/lib/libpkcs7.$(LIB_SUFFIX) \
 	$(DIST)/lib/libcerthi.$(LIB_SUFFIX) \
diff -urN mozilla.orig/security/coreconf/Linux.mk mozilla/security/coreconf/Linux.mk
--- mozilla.orig/security/coreconf/Linux.mk	Sat Oct 27 02:38:08 2001
+++ mozilla/security/coreconf/Linux.mk	Sat Jan 26 23:18:08 2002
@@ -111,5 +111,5 @@
 DSO_LDOPTS		= -shared
 DSO_LDFLAGS		=
 
-# INCLUDES += -I/usr/include -Y/usr/include/linux
+INCLUDES += -I/usr/include `nspr-config --cflags`
 G++INCLUDES		= -I/usr/include/g++
diff -urN mozilla.orig/security/nss/cmd/modutil/manifest.mn mozilla/security/nss/cmd/modutil/manifest.mn
--- mozilla.orig/security/nss/cmd/modutil/manifest.mn	Fri Dec 22 18:08:42 2000
+++ mozilla/security/nss/cmd/modutil/manifest.mn	Sat Jan 26 23:18:48 2002
@@ -54,6 +54,6 @@
 
 DEFINES = -DNSPR20
 
-INCLUDES =
+INCLUDES = -I(CORE_DEPTH)/nss/lib/nss
 
 USE_STATIC_LIBS = 1
diff -urN mozilla.orig/security/nss/lib/ckfw/builtins/Makefile mozilla/security/nss/lib/ckfw/builtins/Makefile
--- mozilla.orig/security/nss/lib/ckfw/builtins/Makefile	Wed May 23 07:55:13 2001
+++ mozilla/security/nss/lib/ckfw/builtins/Makefile	Sat Jan 26 23:19:40 2002
@@ -60,8 +60,7 @@
 EXTRA_LIBS += \
         $(DIST)/lib/libnssckfw.$(LIB_SUFFIX) \
         $(DIST)/lib/libnssb.$(LIB_SUFFIX) \
-	$(PLC_STATIC_LIB) \
-	$(PLDS_STATIC_LIB) \
+	`nspr-config --libs` \
 	$(NULL)
 
 endif
diff -urN mozilla.orig/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile mozilla/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile
--- mozilla.orig/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile	Mon Apr 30 18:40:18 2001
+++ mozilla/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile	Sat Jan 26 23:20:19 2002
@@ -92,8 +92,7 @@
 	$(DIST)/lib/libsoftoken.$(LIB_SUFFIX) \
 	$(CRYPTO_LIB) \
 	$(DIST)/lib/libsecutil.$(LIB_SUFFIX) \
-	$(PLC_STATIC_LIB) \
-	$(PLDS_STATIC_LIB) \
+	`nspr-config --libs` \
 	$(NULL)
 
 endif
diff -urN mozilla.orig/security/nss/lib/smime/manifest.mn mozilla/security/nss/lib/smime/manifest.mn
--- mozilla.orig/security/nss/lib/smime/manifest.mn	Thu Jan 18 21:29:04 2001
+++ mozilla/security/nss/lib/smime/manifest.mn	Sat Jan 26 23:21:03 2002
@@ -70,6 +70,8 @@
 	smimever.c \
 	$(NULL)
 
+INCLUDES = -I$(CORE_DEPTH)/nss/lib/nss
+
 REQUIRES = security dbm
 
 LIBRARY_NAME = smime
diff -urN mozilla.orig/security/nss/lib/ssl/manifest.mn mozilla/security/nss/lib/ssl/manifest.mn
--- mozilla.orig/security/nss/lib/ssl/manifest.mn	Thu Jan 18 21:29:08 2001
+++ mozilla/security/nss/lib/ssl/manifest.mn	Sat Jan 26 23:21:34 2002
@@ -72,6 +72,8 @@
 	nsskea.c \
 	$(NULL)
 
+INCLUDES = $(CORE_DEPTH)/nss/lib/nss
+
 REQUIRES = security dbm
 
 LIBRARY_NAME = ssl
