--- mozilla/accessible/src/base/nsAccessible.cpp.orig	Tue Oct  8 19:49:52 2002
+++ mozilla/accessible/src/base/nsAccessible.cpp	Wed Jun 25 10:59:13 2003
@@ -557,7 +557,7 @@
 
 NS_IMETHODIMP nsAccessible::GetAccId(PRInt32 *aAccId)
 {
-  *aAccId = - NS_REINTERPRET_CAST(PRInt32, (mDOMNode.get()));
+  *aAccId = - (NS_REINTERPRET_CAST(PRPtrdiff, (mDOMNode.get())) & PR_UINT32_MAX);
   return NS_OK;
 }
 
--- mozilla/accessible/src/xul/nsXULTreeAccessible.cpp.orig	Tue Sep 17 05:51:36 2002
+++ mozilla/accessible/src/xul/nsXULTreeAccessible.cpp	Wed Jun 25 11:01:26 2003
@@ -691,7 +691,7 @@
 NS_IMETHODIMP nsXULTreeitemAccessible::GetAccId(PRInt32 *aAccId)
 {
   // Since mDOMNode is same for all tree item, use |this| pointer as the unique Id
-  *aAccId = - NS_REINTERPRET_CAST(PRInt32, this);
+  *aAccId = - (NS_REINTERPRET_CAST(PRPtrdiff, this) & PR_UINT32_MAX);
   return NS_OK;
 }
 
