diff -Nur mozilla/modules/libpref/src/init/all.js mozilla.new/modules/libpref/src/init/all.js
--- mozilla/modules/libpref/src/init/all.js	Thu May  3 05:25:40 2001
+++ mozilla.new/modules/libpref/src/init/all.js	Sun May 13 21:33:38 2001
@@ -497,3 +497,4 @@
 
 
 pref("browser.throbber.url","chrome://navigator-region/locale/region.properties");
+pref("browser.download.dir","~/.gnome-desktop");
\ No newline at end of file
diff -Nur mozilla/profile/defaults/localstore.rdf mozilla.new/profile/defaults/localstore.rdf
--- mozilla/profile/defaults/localstore.rdf	Fri Sep  1 23:53:12 2000
+++ mozilla.new/profile/defaults/localstore.rdf	Sun May 13 21:34:32 2001
@@ -51,8 +51,8 @@
     <NC:persist resource="chrome://navigator/content/navigator.xul#sidebar-panels"/>
   </RDF:Description>
  <RDF:Description about="chrome://navigator/content/navigator.xul#main-window">
-    <width>640</width>
-    <height>480</height>
+    <width>700</width>
+    <height>550</height>
     <screenX>212</screenX>
     <screenY>43</screenY>
   </RDF:Description>
diff -Nur mozilla/xpfe/browser/resources/content/navigator.xul mozilla.new/xpfe/browser/resources/content/navigator.xul
--- mozilla/xpfe/browser/resources/content/navigator.xul	Wed Apr 18 06:40:16 2001
+++ mozilla.new/xpfe/browser/resources/content/navigator.xul	Sun May 13 21:35:35 2001
@@ -51,7 +51,7 @@
         titlemodifier="&mainWindow.titlemodifier;" 
         titlemenuseparator="&mainWindow.titlemodifiermenuseparator;"
         windowtype="navigator:browser" orient="vertical"
-        width="640" height="480"
+	width="700" height="550"
         screenX="10" screenY="10"
         persist="screenX screenY width height sizemode"> 
 
diff -Nur mozilla/xpfe/browser/resources/locale/en-US/region.properties mozilla.new/xpfe/browser/resources/locale/en-US/region.properties
--- mozilla/xpfe/browser/resources/locale/en-US/region.properties	Tue Apr 10 01:13:30 2001
+++ mozilla.new/xpfe/browser/resources/locale/en-US/region.properties	Sun May 13 21:39:55 2001
@@ -1,5 +1,5 @@
 # navigator.properties
-homePageDefault=http://www.mozilla.org/
+homePageDefault=http://www.pld.org.pl/
 shopKeyword=keyword:shop [Product]
 quoteKeyword=keyword:quote [Enter symbol here]
 localKeyword=keyword:zip [Your zip code]
@@ -11,8 +11,9 @@
 #
 # all.js
 #
+browser.startup.homepage=http://www.pld.org.pl/
 browser.startup.homepage=http://www.mozilla.org/
-browser.throbber.url=http://www.mozilla.org/
+browser.throbber.url=http://www.pld.org.pl/
 browser.search.defaulturl=http://search.netscape.com/cgi-bin/search?search=
 
 wallet.Server=http://www.mozilla.org/wallet/tables/
@@ -20,7 +21,7 @@
 
 #config.js
 #
-startup.homepage_override_url=http://www.mozilla.org/mozorg.html
+startup.homepage_override_url=http://www.pld.org.pl/
 
 # search-panel.properties
 #   Localizable URLs
