diff -Nru mozilla.orig/config/config.mk mozilla/config/config.mk
--- mozilla.orig/config/config.mk	Sat Dec 28 02:12:56 2002
+++ mozilla/config/config.mk	Wed Feb 12 12:50:40 2003
@@ -178,7 +178,7 @@
 else
 NSS_LIBS	= \
 	$(LIBS_DIR) \
-	$(DIST)/lib/$(LIB_PREFIX)crmf.$(LIB_SUFFIX) \
+	/usr/lib/libcrmf.a \
 	-lsmime3 \
 	-lssl3 \
 	-lnss3 \
@@ -187,11 +187,6 @@
 endif
 
 NSS_DEP_LIBS	= \
-	$(DIST)/lib/$(LIB_PREFIX)crmf.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)smime3$(DLL_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)ssl3$(DLL_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)nss3$(DLL_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)softokn3$(DLL_SUFFIX) \
 	$(NULL)
 
 MOZ_UNICHARUTIL_LIBS = $(DIST)/lib/$(LIB_PREFIX)unicharutil_s.$(LIB_SUFFIX)
diff -Nru mozilla.orig/security/manager/Makefile.in mozilla/security/manager/Makefile.in
--- mozilla.orig/security/manager/Makefile.in	Sat Dec 28 06:50:17 2002
+++ mozilla/security/manager/Makefile.in	Wed Feb 12 12:52:48 2003
@@ -110,29 +110,6 @@
 else
 	cd $(DIST)/lib; cp -f $(LIB_PREFIX)mozdbm_s.$(LIB_SUFFIX) $(LIB_PREFIX)dbm.$(LIB_SUFFIX); $(RANLIB) $(LIB_PREFIX)dbm.$(LIB_SUFFIX)
 endif
-	$(MAKE) -C $(topsrcdir)/security/nss/lib $(DEFAULT_GMAKE_FLAGS)
-ifndef DISABLE_DIST_GRE
-	$(INSTALL) -m 755 $(DIST)/lib/$(LOADABLE_ROOT_MODULE) $(GRE_DIST)
-	$(INSTALL) -m 755 $(DIST)/lib/$(SOFTOKEN3_LIB) $(GRE_DIST)
-	$(INSTALL) -m 755 $(DIST)/lib/$(NSS3_LIB) $(GRE_DIST)
-	$(INSTALL) -m 755 $(DIST)/lib/$(SSL3_LIB) $(GRE_DIST)
-	$(INSTALL) -m 755 $(DIST)/lib/$(SMIME3_LIB) $(GRE_DIST)
-ifdef HAVE_FREEBL_MODULES
-	$(INSTALL) -m 755 $(DIST)/lib/$(FREEBL_PURE32_MODULE) $(GRE_DIST)
-	$(INSTALL) -m 755 $(DIST)/lib/$(FREEBL_HYBRID_MODULE) $(GRE_DIST)
-endif
-endif
-ifndef _SKIP_OLD_GRE_INSTALL
-	$(INSTALL) -m 755 $(DIST)/lib/$(LOADABLE_ROOT_MODULE) $(DIST)/bin
-	$(INSTALL) -m 755 $(DIST)/lib/$(SOFTOKEN3_LIB) $(DIST)/bin
-	$(INSTALL) -m 755 $(DIST)/lib/$(NSS3_LIB) $(DIST)/bin
-	$(INSTALL) -m 755 $(DIST)/lib/$(SSL3_LIB) $(DIST)/bin
-	$(INSTALL) -m 755 $(DIST)/lib/$(SMIME3_LIB) $(DIST)/bin
-ifdef HAVE_FREEBL_MODULES
-	$(INSTALL) -m 755 $(DIST)/lib/$(FREEBL_PURE32_MODULE) $(DIST)/bin
-	$(INSTALL) -m 755 $(DIST)/lib/$(FREEBL_HYBRID_MODULE) $(DIST)/bin
-endif
-endif
 	$(MAKE) -C boot $@
 	$(MAKE) -C ssl $@
 	$(MAKE) -C pki $@
diff -Nru mozilla.orig/security/manager/ssl/src/Makefile.in mozilla/security/manager/ssl/src/Makefile.in
--- mozilla.orig/security/manager/ssl/src/Makefile.in	Sat Jan 18 15:02:51 2003
+++ mozilla/security/manager/ssl/src/Makefile.in	Wed Feb 12 12:53:51 2003
@@ -111,8 +111,7 @@
 
 # Use local includes because they are inserted before INCLUDES
 # so that Mozilla's nss.h is used, not glibc's
-LOCAL_INCLUDES += -I$(DIST)/public/nss \
-                  -I$(DIST)/private/nss \
+LOCAL_INCLUDES += -I/usr/include/nss \
                   $(NULL)
 
 EXTRA_DSO_LDOPTS += \
