--- mozilla/view/src/nsViewManager.cpp.orig	2003-12-17 22:05:37.000000000 +0100
+++ mozilla/view/src/nsViewManager.cpp	2004-05-07 10:53:08.726357344 +0200
@@ -2879,7 +2879,8 @@
   // code still does the right thing. Yay for the display list based
   // approach!)
 
-  OptimizeDisplayList(&displayList, nsRegion(r), finalTransparentRect, opaqueRegion, PR_TRUE);
+  const nsRegion region(r);
+  OptimizeDisplayList(&displayList, region, finalTransparentRect, opaqueRegion, PR_TRUE);
 
   PRBool anyUnscrolledViews = PR_FALSE;
   PRBool anyUnblittableViews = PR_FALSE;
@@ -3285,7 +3286,8 @@
 
   // Paint the view. The clipping rect was set above set don't clip again.
   //aView->Paint(*localcx, trect, NS_VIEW_FLAG_CLIP_SET, result);
-  RenderViews(view, *localcx, nsRegion(trect), PR_FALSE);
+  const nsRegion region(trect);
+  RenderViews(view, *localcx, region, PR_FALSE);
 
   NS_RELEASE(localcx);
 
