diff -Nur mozilla.orig/embedding/config/basebrowser-unix mozilla/embedding/config/basebrowser-unix
--- mozilla.orig/embedding/config/basebrowser-unix	Fri Jun  8 14:05:26 2001
+++ mozilla/embedding/config/basebrowser-unix	Sat Jul  7 06:39:04 2001
@@ -163,13 +163,13 @@
 
 ; psm2
 ; Optional - only if you need PSM2 support
-; components/libpipnss.so
-; components/pipnss.xpt
-; chrome/pipnss.jar
-; components/libpippki.so
-; components/pippki.xpt
-; chrome/pippki.jar
-; defaults/pref/security-prefs.js
+components/libpipnss.so
+components/pipnss.xpt
+chrome/pipnss.jar
+components/libpippki.so
+components/pippki.xpt
+chrome/pippki.jar
+defaults/pref/security-prefs.js
 
 ; rdf
 components/librdf.so
