--- mozilla/modules/plugin/nglsrc/nsPluginsDirUnix.cpp.orig	Wed Nov  8 04:10:37 2000
+++ mozilla/modules/plugin/nglsrc/nsPluginsDirUnix.cpp	Mon Jan  8 10:36:43 2001
@@ -178,10 +178,10 @@
     // work fine.
 
     tempSpec.type = PR_LibSpec_Pathname;
-    tempSpec.value.pathname = "libXt.so";
+    tempSpec.value.pathname = "libXt.so.6";
     handle = PR_LoadLibraryWithFlags(tempSpec, PR_LD_NOW|PR_LD_GLOBAL);
 
-    tempSpec.value.pathname = "libXext.so";
+    tempSpec.value.pathname = "libXext.so.6";
     handle = PR_LoadLibraryWithFlags(tempSpec, PR_LD_NOW|PR_LD_GLOBAL);
 
     libSpec.type = PR_LibSpec_Pathname;
