--- mozilla/directory/c-sdk/ldap/libraries/libssldap/Makefile.in.orig	Fri May 17 08:54:00 2002
+++ mozilla/directory/c-sdk/ldap/libraries/libssldap/Makefile.in	Wed Aug 28 11:37:29 2002
@@ -53,9 +53,8 @@
 LOCAL_INCLUDES  = -I$(PUBLIC)/nspr
 INCLUDES	+= -I$(DISTHDIR) -I$(HDIR) -I$(INSTALLDIR)/include \
 		   -I$(DIST)/include \
-		   -I$(DIST)/public/security \
-		   -I$(DIST)/public/nss \
-                   -I$(DIST)/include/nspr
+		   -I/usr/include/nss \
+                   -I/usr/include/nspr
 DEFINES		+= $(DEFS)
 
 PLATFORMCFLAGS	= -DUSE_WAITPID -DNEEDPROTOS
--- mozilla/configure.in~	2004-05-23 23:31:57.220256656 +0200
+++ mozilla/configure.in	2004-05-23 23:33:27.248570272 +0200
@@ -5902,7 +5902,7 @@
         mkdir "directory/c-sdk/ldap"    
     fi
 
-    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --without-nss --with-mozilla"
+    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --with-nss --with-mozilla"
     if test -z "$MOZ_DEBUG"; then
         ac_configure_args="$ac_configure_args --disable-debug"
     fi
--- mozilla/configure~	2004-05-23 23:31:57.294245408 +0200
+++ mozilla/configure	2004-05-23 23:33:38.553851608 +0200
@@ -17768,7 +17768,7 @@
         mkdir "directory/c-sdk/ldap"    
     fi
 
-    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --without-nss --with-mozilla"
+    ac_configure_args="$_SUBDIR_CONFIG_ARGS --prefix=$MOZ_BUILD_ROOT/dist --with-dist-prefix=$MOZ_BUILD_ROOT/dist --with-nss --with-mozilla"
     if test -z "$MOZ_DEBUG"; then
         ac_configure_args="$ac_configure_args --disable-debug"
     fi
